OloEditor/src/EditorLayer.cpp

```
@@ -82,9 +82,18 @@ namespace OloEngine
         // Cancel any ongoing build
         m_BuildCancelRequested.store(true);
 
-        // Wait for build to complete if running (spin-wait with yield)
+        // Wait for build to complete with a bounded timeout so the destructor cannot hang forever
+        constexpr auto timeout = std::chrono::seconds(5);
```

Cppcheck (2.20.0)

[style] The function 'GetType' is never used.

(unusedFunction)